proof.go

```diff
@@ -118,6 +119,9 @@ func (proof Proof) VerifyNamespace(nth Hasher, nID namespace.ID, data []namespac
 			gotLeafHashes = append(gotLeafHashes, hashLeafFunc(gotLeaf.Bytes()))
 		}
 	}
+	if !proof.IsOfAbsence() && len(gotLeafHashes) != (proof.End()-proof.Start()) {
+		return false, ErrMissingData
+	}
```

Comment:
without this, the verification might panic (e.g. when removing a leaf before calling `VerifyNamespace`.